azurerm/resource_arm_autoscale_setting_test.go

```diff
@@ -0,0 +1,831 @@
+
+      frontend_ip_configuration { 
+        name                 = "PublicIPAddress" 
+        public_ip_address_id = "${azurerm_public_ip.test.id}" 
+      } 
+    } 
+    
+    resource "azurerm_lb_backend_address_pool" "bpepool" { 
+      name                = "BackEndAddressPool" 
+      resource_group_name = "${azurerm_resource_group.test.name}" 
+      loadbalancer_id     = "${azurerm_lb.test.id}" 
+    } 
+    
+    resource "azurerm_lb_nat_pool" "lbnatpool" { 
+      count                          = 3 
+      resource_group_name            = "${azurerm_resource_group.test.name}" 
+      name                           = "ssh" 
+      loadbalancer_id                = "${azurerm_lb.test.id}" 
+      protocol                       = "Tcp" 
+      frontend_port_start            = 50000 
+      frontend_port_end              = 50119 
+      backend_port                   = 22 
+      frontend_ip_configuration_name = "PublicIPAddress" 
+    } 
+    
+    resource "azurerm_virtual_machine_scale_set" "test" { 
+      name                = "myTestScaleset-1" 
+      location            = "${azurerm_resource_group.test.location}" 
+      resource_group_name = "${azurerm_resource_group.test.name}" 
+      upgrade_policy_mode = "Manual" 
+    
+      sku { 
+        name     = "Standard_A0" 
+        tier     = "Standard" 
+        capacity = 1 
+      } 
+    
+      storage_profile_image_reference { 
+        publisher = "Canonical" 
+        offer     = "UbuntuServer" 
+        sku       = "14.04.2-LTS" 
+        version   = "latest" 
+      } 
+    
+      storage_profile_os_disk { 
+        name              = "" 
+        caching           = "ReadWrite" 
+        create_option     = "FromImage" 
+        managed_disk_type = "Standard_LRS" 
+      } 
+    
+      storage_profile_data_disk { 
+        lun            = 0 
+        caching        = "ReadWrite" 
+        create_option  = "Empty" 
+        disk_size_gb   = 10  
+      } 
+    
+      os_profile { 
+        computer_name_prefix = "testvm" 
+        admin_username       = "myadmin" 
+        admin_password       = "${var.adminPwd}" 
+      } 
+    
+      os_profile_linux_config { 
+        disable_password_authentication = false 
+      } 
+    
+      network_profile { 
+        name    = "terraformnetworkprofile" 
+        primary = true 
+    
+        ip_configuration { 
+          name                                   = "TestIPConfiguration" 
+          subnet_id                              = "${azurerm_subnet.test.id}" 
+          load_balancer_backend_address_pool_ids = ["${azurerm_lb_backend_address_pool.bpepool.id}"] 
+          load_balancer_inbound_nat_rules_ids    = ["${element(azurerm_lb_nat_pool.lbnatpool.*.id, count.index)}"] 
+        } 
+      } 
+    
+      tags { 
+        environment = "staging" 
+      } 
+    } 
+
+    resource "azurerm_autoscale_setting" "test" {
+      name                = "myAutoscaleSetting"
```

Comment:
The resource group does have a unique name, any reason for individual resources to have unique names too?